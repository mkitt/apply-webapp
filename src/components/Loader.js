// @flow
import React from 'react'
import glamorous from 'glamorous'
import { animateRotate, easeInOutCubic } from '../constants/style'

const View = glamorous.div({
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  borderRadius: '50%',
})

const Busy = glamorous.div({
  width: 16,
  height: 16,
  marginRight: 8,
  borderRadius: '50%',
  borderWidth: 2,
  borderStyle: 'dashed',
  borderColor: '#404040',
  animation: `${animateRotate} 1.03333s infinite ${easeInOutCubic}`,
})

type Props = {
  children: React.Element<*>,
}

export default ({ children, ...props }: Props) => (
  <View {...props}>
    <Busy marginRight={100} />
    {children}
  </View>
)
