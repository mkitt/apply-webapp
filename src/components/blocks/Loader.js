// @flow
import React, { type Node } from 'react'
import glamorous from 'glamorous'
import { animateRotate, easeInOutCubic } from '../../constants/style'

type Props = {
  children: Node,
}

export default ({ children, ...props }: Props) => (
  <View {...props}>
    <Busy />
    {children}
  </View>
)

// -------------------------------------

const View = glamorous.div({
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  borderRadius: '50%',
})

const Busy = glamorous.div({
  width: 16,
  height: 16,
  marginRight: 8,
  borderRadius: '50%',
  borderWidth: 2,
  borderStyle: 'dashed',
  borderColor: '#333',
  animation: `${animateRotate} 1.03333s infinite ${easeInOutCubic}`,
})
