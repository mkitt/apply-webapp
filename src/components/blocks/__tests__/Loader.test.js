import React from 'react'
import renderer from 'react-test-renderer'
import Loader from '../Loader'

describe('<Loader />', () => {
  it('shows the correct snapshot tree for the <Loader> block', () => {
    const component = (
      <Loader>Loading Loader...</Loader>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
