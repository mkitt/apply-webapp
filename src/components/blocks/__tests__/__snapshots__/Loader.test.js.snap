// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> shows the correct snapshot tree for the <Loader> block 1`] = `
.css-0,
[data-css-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 50%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-1,
[data-css-1] {
  width: 16px;
  height: 16px;
  margin-right: 8px;
  border-radius: 50%;
  border-width: 2px;
  border-style: dashed;
  border-color: #4d4d4d;
  animation: ANIMATE_ROTATE_1a8hhxx 1.03333s infinite cubic-bezier(0.645, 0.045, 0.355, 1);
  -webkit-animation: ANIMATE_ROTATE_1a8hhxx 1.03333s infinite cubic-bezier(0.645, 0.045, 0.355, 1);
}

<div
  className="css-0"
>
  <div
    className="css-1"
  />
  Loading Loader...
</div>
`;
